dockerfiles/init/modules/openshift/files/scripts/ocp.sh

```diff
@@ -148,15 +158,46 @@ run_ocp() {
 }
 
 deploy_che_to_ocp() {
-    #Repull init image only if IMAGE_PULL_POLICY is set to Always
-    if [ $IMAGE_PULL_POLICY == "Always" ]; then
-        docker pull "$IMAGE_INIT"
+    #Only generate scripts and config files if deploy_che.sh does not exist in same folder
+    if [ ! -f deploy_che.sh ]; then
+      #Repull init image only if IMAGE_PULL_POLICY is set to Always
+      if [ $IMAGE_PULL_POLICY == "Always" ]; then
+          docker pull "$IMAGE_INIT"
+      fi
+      docker run -t --rm -v /var/run/docker.sock:/var/run/docker.sock -v "${CONFIG_DIR}":/data -e IMAGE_INIT="$IMAGE_INIT" -e CHE_MULTIUSER="$CHE_MULTIUSER" eclipse/che-cli:${CHE_IMAGE_TAG} destroy --quiet --skip:pull --skip:nightly
+      docker run -t --rm -v /var/run/docker.sock:/var/run/docker.sock -v "${CONFIG_DIR}":/data -e IMAGE_INIT="$IMAGE_INIT" -e CHE_MULTIUSER="$CHE_MULTIUSER" eclipse/che-cli:${CHE_IMAGE_TAG} config --skip:pull --skip:nightly
+      cd "${CONFIG_DIR}/instance/config/openshift/scripts/"
+    fi
+    if $CHE_REMOVE_PROJECT; then
+      remove_che_from_ocp
     fi
-    docker run -t --rm -v /var/run/docker.sock:/var/run/docker.sock -v "${CONFIG_DIR}":/data -e IMAGE_INIT="$IMAGE_INIT" -e CHE_MULTIUSER="$CHE_MULTIUSER" eclipse/che-cli:${CHE_IMAGE_TAG} destroy --quiet --skip:pull --skip:nightly
-    docker run -t --rm -v /var/run/docker.sock:/var/run/docker.sock -v "${CONFIG_DIR}":/data -e IMAGE_INIT="$IMAGE_INIT" -e CHE_MULTIUSER="$CHE_MULTIUSER" eclipse/che-cli:${CHE_IMAGE_TAG} config --skip:pull --skip:nightly
-    cd "${CONFIG_DIR}/instance/config/openshift/scripts/"
     bash deploy_che.sh ${DEPLOY_SCRIPT_ARGS}
     wait_until_server_is_booted
+    CHE_DEPLOYED=true
+    if $CHE_REPLACE_STACKS; then
+      replace_che_stacks_on_ocp
+    fi
+}
+
+replace_che_stacks_on_ocp() {
+  if ! $CHE_DEPLOY || $CHE_DEPLOYED; then
+    echo "[CHE] Checking if project \"${CHE_OPENSHIFT_PROJECT}\" exists before replacing stacks..."
+    if $OC_BINARY get project "${CHE_OPENSHIFT_PROJECT}" &> /dev/null; then
+      echo "[CHE] Replacing Che Stacks."
+      if [ ! -f replace_stacks.sh ]; then
```

Comment:
again we should not use local scripts.

Reply:
How would you propose to debug scripts without having the ability to make changes locally. Also, there is no way to change where the replace_stack.sh pulls it's stack definitions from when using the generated scripts. I could redo this script so that there is a command line argument `--local-scripts` that will use local script in lieu of generated. This means that unless user specifically wants to use local scripts it will default to generated scripts as it does currently.

Reply:
build image -> define pull policy  that is how you debug
replace stacks it is obsolete script which does not work in multiuser and lost any sense in che6 and will be deleted soon.

Reply:
If you make modifications to a script in "${CONFIG_DIR}/instance/config/openshift/scripts/" it will replace any scripts I modify within that folder such as deploy_che.sh . I think the ability to use your own scripts instead of the ones in the docker should be an option as users might need to patch the scripts to get things running on their infrastructure. I would be ok with having a variable that indicates to the docker run not to replace existing files if they exist. This would not be the default though but would allow something like the following.

export CONFIG_DIR=$(pwd)
export REPLACE_EXISTING_CONF=false #default true

```
docker run -t --rm -v /var/run/docker.sock:/var/run/docker.sock -v "${CONFIG_DIR}":/data -e IMAGE_INIT="$IMAGE_INIT" -e CHE_MULTIUSER="$CHE_MULTIUSER" -e REPLACE_EXISTING_CONF="$REPLACE_EXISTING_CONF" eclipse/che-cli:${CHE_IMAGE_TAG} destroy --quiet --skip:pull --skip:nightly
docker run -t --rm -v /var/run/docker.sock:/var/run/docker.sock -v "${CONFIG_DIR}":/data -e IMAGE_INIT="$IMAGE_INIT" -e CHE_MULTIUSER="$CHE_MULTIUSER" -e REPLACE_EXISTING_CONF="$REPLACE_EXISTING_CONF" eclipse/che-cli:${CHE_IMAGE_TAG} config --skip:pull --skip:nightly
```

Reply:
Are you stating that the script files should be edited in the image and then commit the image to override the existing? This doesn't seem like the easiest solution as I would need to ssh into the image to get an editor that is not terminal to edit the file.